### PR TITLE
refactor: change handling of config files

### PR DESCRIPTION
**Pull request by kleyow**

### CODE REVIEW

**Comment by kleyow** [comment]
> knexfile.ts lives in the config directory, which will often be shadowed by k8s config dir mounts

Can you explain this one Lewis? knexfile live's in `config/` in other projects so what becomes the problem in this case?

File: config/convictFileConfig.ts [hunk 1]
```diff
@@ -0,0 +1,166 @@
...
+    CREATE_TIMEOUT_MILLIS: number;
+    DESTROY_TIMEOUT_MILLIS: number;
+    IDLE_TIMEOUT_MILLIS: number;
+    REAP_INTERVAL_MILLIS: number;
+    CREATE_RETRY_INTERVAL_MILLIS: number;
+    USE_NULL_AS_DEFAULT?: boolean;
+  }
+}
+
+const ConvictFileConfig = Convict<FileConfig>({
```
Comment (eoln) [inline config/convictFileConfig.ts:59]: what about getting some parameters not from the config file but also from the environment? - 12fa suggest it - especially for security properties like 'passwords' or basic like 'port/host'

**Review by eoln — APPROVED** [review]
lgtm

File: config/convictFileConfig.ts (hunk 1 above)
Comment (kleyow) [reply config/convictFileConfig.ts:59]: Not familiar on how to do that with convict. Let me do that in a followup PR.

**Pull request merged**